### ISSUE TITLE
[BUG] [vsphere] Error waiting for cluster to be updated: unexpected state 'waiting', wanted target 'active, provisioning, pending'

### ISSUE THREAD

**Issue by dbsanfte** [rancher/terraform-provider-rancher2#1353]
### Rancher Server Setup
- Rancher version: 2.8.3
- Installation option (Docker install/Helm Chart): Helm Chart
   - k3s at v1.28.6+k3s2
- Proxy/Cert Details: Self signed cert

### Information about the Cluster
- Kubernetes version: v1.28.8-rancher1-1
- Cluster Type (Local/Downstream): Downstream
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider): Hosted, vsphere

### User Information
- What is the role of the user logged in? Admin

### Provider Information
- What is the version of the Rancher v2 Terraform Provider in use? 4.1.0
- What is the version of Terraform in use? 1.8.4

### Describe the bug
Provisioning in-cluster resources with the Rancher provider occasionally fails with the below error while the Rancher 2 provider is waiting on the cluster to be ready (via the cluster_sync resource).

### To Reproduce
* Create a vsphere cluster with an ESX that takes a fair amount of time to provision the hosts (about 25mins in my case)

### Actual Result
```
Error: [ERROR] waiting for cluster (c-4tst9) to be updated: unexpected state 'waiting', wanted target 'active, provisioning, pending'. last error: %!s(<nil>)

  with rancher2_cluster.vsphere_backend,
  on backend.tf line 1, in resource "rancher2_cluster" "vsphere_backend":
   1: resource "rancher2_cluster" "vsphere_backend" {
```

### Expected Result
The provider should wait for the cluster to be ready, no matter what state it is in.

**Comment by dbsanfte** [comment]
This is still an issue with Rancher 2.10.0

```
Error: [ERROR] waiting for cluster (c-j7wz4) to be updated: unexpected state 'waiting', wanted target 'active, provisioning, pending'. last error: %!s(<nil>)
```

**Comment by dbsanfte** [comment]
Created a PR to fix this, see above.

**Comment by matttrach** [comment]
This only references older/unsupported versions of Rancher and the provider, as such I am closing this issue, but feel free to open a new issue if this is still a valid problem.
